scripts/build_podspec.sh

```diff
@@ -0,0 +1,133 @@
+#!/bin/bash
+##===----------------------------------------------------------------------===##
+##
+## This source file is part of the SwiftNIO open source project
+##
+## Copyright (c) 2017-2019 Apple Inc. and the SwiftNIO project authors
+## Licensed under Apache License v2.0
+##
+## See LICENSE.txt for license information
+## See CONTRIBUTORS.txt for the list of SwiftNIO project authors
+##
+## SPDX-License-Identifier: Apache-2.0
+##
+##===----------------------------------------------------------------------===##
+
+set -eu
+
+function usage() {
+  echo "$0 [-u] version nio_version"
+  echo
+  echo "OPTIONS:"
+  echo "  -u: Additionally upload the podspec"
+  echo "  -f: Skip over all targets before the specified target"
+}
+
+upload=false
+skip_until=""
+while getopts ":u" opt; do
+  case $opt in
+    u)
+      upload=true
+      ;;
+    f)
+      skip_until="$OPTARG"
+      ;;
+    \?)
+      usage
+      exit 1
+      ;;
+  esac
+done
+shift "$((OPTIND-1))"
+
+if [[ $# -eq 0 ]]; then
+  echo "Must provide target version"
+  exit 1
+fi
+
+version=$1
+
+# Current SwiftNIO Version to add as dependency in the .podspec
+nio_version=$2
+newline=$'\n'
+
+here="$( cd "$( dirname "${BASH_SOURCE[0]}" )" && pwd )"
+tmpdir=$(mktemp -d /tmp/.build_podspecsXXXXXX)
+
+echo "Building podspec in $tmpdir"
+
+targets=( $("${here}/list_topsorted_dependencies.sh" -l -r | sed 's/^NIO/SwiftNIO/') )
+
+for target in "${targets[@]}"; do
+
+  if [[ -n "$skip_until" && "$target" != "$skip_until" ]]; then
+    echo "Skipping $target"
+    continue
+  elif [[ "$skip_until" == "$target" ]]; then
+    skip_until=""
+  fi
+
+  echo "Building podspec for $target"
+
+  dependencies=()
+
+  while read -r raw_dependency; do
+    if [[ "$raw_dependency" =~ ^CNIO ]]; then
+      dependencies+=( "${newline}  s.dependency '$raw_dependency', s.version.to_s" )
+    else
+      dependencies+=( "${newline}  s.dependency '$raw_dependency', '$nio_version'" )
+    fi
+  done < <("${here}/list_topsorted_dependencies.sh" -d "${target#Swift}" | sed 's/^NIO/SwiftNIO/')
+
+
+  # C++ specific podspec settings
+  libraries=""
+  xcconfig=""
+  public_header_files=""
+
+  if [ "$target" == "CNIOBoringSSL" ] || [ "$target" == "CNIOBoringSSLShims" ]; then
+    libraries="s.libraries = 'c++'"
+    xcconfig="s.xcconfig = { 'HEADER_SEARCH_PATHS' => 'Sources/${target}/include', 'CLANG_CXX_LANGUAGE_STANDARD' => 'c++11', }"
+    public_header_files="s.public_header_files = 'Sources/${target}/include/*.h'"
+  fi
+
+  cat > "${tmpdir}/${target}.podspec" <<- EOF
+Pod::Spec.new do |s|
+  s.name = '$target'
```

Comment:
doesn't this need a `Swift` in front? The pod names should be `SwiftNIOSSL` and `CNIOBoringSSL` to stay in line with SwiftNIO (ie. if the target starts with `NIO`, prepend a `Swift` to make it `SwiftNIO...`; if the target starts with a `C` or anything else, just leave it as is).

Reply:
Correct me if I'm wrong, I believe this one does need `Swift` in front of it, whereas below for the `module_name` we strip `Swift` from it.

For the SwiftNIOSSL specific podspec we would have:
```
s.name = 'SwiftNIOSSL'
s.module_name = 'NIOSSL'
```

Similarly, for SwiftNIO we have:
```
s.name = 'SwiftNIO'
s.module_name = 'NIO'
```

What do you think?

Reply:
yes, that's correct

Reply:
@Jake-Prickett is that already what's being generated? if yes, I'm happy :)

Reply:
Sure is! 👍🏼

Reply:
thank you, let's get this merged then.